src/systems/a320_systems/src/flaps_computer.rs

```diff
@@ -0,0 +1,1347 @@
+use systems::simulation::{
+    Read, SimulationElement, SimulationElementVisitor, SimulatorReader, SimulatorWriter,
+    UpdateContext, Write,
+};
+
+use uom::si::{angle::degree, f64::*, velocity::knot};
+
+//The different flaps configurations
+#[derive(Debug, Copy, Clone, PartialEq)]
+enum FlapsConf {
+    Conf0 = 0,
+    Conf1 = 1,
+    Conf1F = 2,
+    Conf2 = 3,
+    Conf3 = 4,
+    ConfFull = 5,
+}
+
+impl From<usize> for FlapsConf {
+    fn from(value: usize) -> Self {
+        match value {
+            0 => FlapsConf::Conf0,
+            1 => FlapsConf::Conf1,
+            2 => FlapsConf::Conf1F,
+            3 => FlapsConf::Conf2,
+            4 => FlapsConf::Conf3,
+            _ => FlapsConf::ConfFull,
+        }
+    }
+}
+
+//A struct to read the handle position
+struct FlapsHandle {
+    handle_position: usize,
+    old_handle_position: usize,
+}
+
+impl FlapsHandle {
+    fn new() -> Self {
+        Self {
+            handle_position: 0,
+            old_handle_position: 0,
+        }
+    }
+
+    fn signal_new_position(&self) -> Option<(usize, usize)> {
+        if self.handle_position == self.old_handle_position {
+            match self.handle_position {
+                1 => Some((1, 1)),
+                _ => None,
+            }
+        } else {
+            Some((self.old_handle_position, self.handle_position))
+        }
+    }
+
+    fn equilibrate_old_and_current(&mut self) {
+        self.old_handle_position = self.handle_position;
+    }
```

Comment:
**suggestion**: Consider moving this line into `read`, above the `self.handle_position = reader.read("FLAPS_HANDLE_INDEX");`. It should be equivalent and reduces the need for this function.
**suggestion**: We tend to use `previous` to indicate the value of the previous frame, not `old`. Consider using `previous`.

Reply:
So what exactly is the "spacing rule"?
```rust
struct MyStruct {
...
}
impl MyS {
...
}

```

without space for any struct?

Reply:
Both suggestions implemented in commit `8cc666e`